### ISSUE TITLE
rasterTools

### ISSUE THREAD

**Issue by EhrmannS** [Robinlovelace/geocompr#265]
Hi, I have now reached a point with [rasterTools](https://github.com/EhrmannS/rasterTools) where I am confident to share it with a more public audience. Is there any way this would fit into geocompr? (it's not yet on CRAN...)

You can find lots of already working examples in the [usecases](https://github.com/EhrmannS/rasterTools/blob/master/vignettes/usecases.Rmd) vignette.

**Comment by Robinlovelace** [comment]
As with the mapdeck issue #264 what are time-lines for submitting to CRAN. First thought from me is: how does it compare with **raster**? Imagine it's a different set of tools, intriguing.

**Comment by EhrmannS** [comment]
The major difference would be the way it is used, instead of writing the typical "scripts" that can be hard to follow, one would put together an "algorithm" of operations that should be carried out on a certain object. For instance to determine patches in a raster with continuous values one would define the algorithm
```
get_patches <- list(list(operator = "rBinarise", thresh = 30),
                    list(operator = "rPatches")
```
and use it to modify a continuous object (alternatively the operators can also be used in a magrittr-pipeline).
```
myInput <- rtData$continuous
myPatches <- modify(input = myInput, by = get_patches, sequential = TRUE)
```
This allows to also prototype more complex algorithms that can readily be used on any continuous raster. The different tools are then operators to modify rasters (all kind of spatial operations), obtain earth observation data or compute landscape metrics (and in the future also to generate spatial patterns).

I am preparing it for CRAN atm, writing tests and make sure everything works as intended.

**Comment by Robinlovelace** [comment]
Thanks for the description. FYI it installed and seems to work fine on my laptop. I'll have a read of the usecases also. Assuming all goes to plan with the CRAN submission and that it opens-up new options I see no reason not to include it, but may not be for a few months as we're all hands on deck finishing the book atm.

**Comment by Robinlovelace** [comment]
Closing for now because we're preparing for the book's publication. Happy to include info on this package when it's on CRAN in preparation for a second edition. Also commented here: https://github.com/Robinlovelace/geocompr/issues/12
